presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -150,38 +153,33 @@ else if (node.getRelation() instanceof Table) {
-                check(!querySpecification.getOrderBy().isPresent(), node, "ORDER BY is not supported in SHOW STATS SELECT clause");
-                check(!querySpecification.getLimit().isPresent(), node, "LIMIT is not supported by SHOW STATS SELECT clause");
-                check(!querySpecification.getHaving().isPresent(), node, "HAVING is not supported in SHOW STATS SELECT clause");
-                check(!querySpecification.getGroupBy().isPresent(), node, "GROUP BY is not supported in SHOW STATS SELECT clause");
-                check(!querySpecification.getSelect().isDistinct(), node, "DISTINCT is not supported by SHOW STATS SELECT clause");
-
-                List<SelectItem> selectItems = querySpecification.getSelect().getSelectItems();
-                check(selectItems.size() == 1 && selectItems.get(0) instanceof AllColumns, node, "Only SELECT * is supported in SHOW STATS SELECT clause");
-
-                querySpecification.getWhere().ifPresent((expression) -> validateShowStatsWhereExpression(expression, node));
-            }
+            Optional<FilterNode> filterNode = searchFrom(plan.getRoot())
+                    .where(FilterNode.class::isInstance)
+                    .findSingle();
+
+            check(!filterNode.isPresent(), node, "Only predicates that can be pushed down are supported in the SHOW STATS WHERE clause");
+            check(querySpecification.getFrom().isPresent(), node, "There must be exactly one table in query passed to SHOW STATS SELECT clause");
+            check(querySpecification.getFrom().get() instanceof Table, node, "There must be exactly one table in query passed to SHOW STATS SELECT clause");
+            check(!query.getWith().isPresent(), node, "WITH is not supported by SHOW STATS SELECT clause");
+            check(!querySpecification.getOrderBy().isPresent(), node, "ORDER BY is not supported in SHOW STATS SELECT clause");
+            check(!querySpecification.getLimit().isPresent(), node, "LIMIT is not supported by SHOW STATS SELECT clause");
+            check(!querySpecification.getHaving().isPresent(), node, "HAVING is not supported in SHOW STATS SELECT clause");
+            check(!querySpecification.getGroupBy().isPresent(), node, "GROUP BY is not supported in SHOW STATS SELECT clause");
+            check(!querySpecification.getSelect().isDistinct(), node, "DISTINCT is not supported by SHOW STATS SELECT clause");
+
+            List<SelectItem> selectItems = querySpecification.getSelect().getSelectItems();
+            check(selectItems.size() == 1 && selectItems.get(0) instanceof AllColumns, node, "Only SELECT * is supported in SHOW STATS SELECT clause");
+
+            querySpecification.getWhere().ifPresent((expression) -> validateShowStatsWhereExpression(expression, node));
         }
 
         private void validateShowStatsWhereExpression(Expression expression, ShowStats node)
```

Comment:
I think this method is no longer needed

Reply:
One example use of this function is in #11545, it rejects query "SHOW STATS FOR (SELECT * FROM nation_partitioned WHERE regionkey BETWEEN 1 AND 3+3)".   Suppose  expression "3+3" should be supported, the question becomes "Should we get rid of ALLOWED_SHOW_STATS_WHERE_EXPRESSION_TYPES in ShowStatsRewrite.java?"  @mbasmanova

Reply:
@kokosing Grzegorz, why do you think this method is not needed?

Reply:
If whole predicate can be pushed down to connector (and there is no `FilterNode` here https://github.com/prestodb/presto/pull/11641/commits/0ae180b1968dc7568d51d32d8632588101b489c8#diff-bba77bbf14976e225b69bac5aa20c899R169), then in my opinion there is no much sense to validate predicate. In case there is something unsupported then it won't be pushed down.

Reply:
@kokosing That's a valid point. Thanks for explaining.